source/common/stats/stat_data_allocator_impl.h

```diff
@@ -87,6 +87,32 @@ template <class StatData> class CounterImpl : public Counter, public MetricImpl
   StatDataAllocatorImpl<StatData>& alloc_;
 };
 
+/**
+ * Null counter implementation.
+ * No-ops on all calls and requires no underlying metric or data.
+ */
+class NullCounterImpl : public Counter {
+public:
+  NullCounterImpl() {}
+  ~NullCounterImpl() {}
+  // Stats::Metric
+  const std::string name() const override { return ""; }
+  // Stats::MetricImpl
+  const std::string& tagExtractedName() const override { return tag_extracted_name_; }
+  const std::vector<Tag>& tags() const override { return tags_; }
+  // Stats::Counter
+  void add(uint64_t amount) override { UNREFERENCED_PARAMETER(amount); }
```

Comment:
nit: you can just remove the parameter name from the signature and the unreferenced parameter warning goes away.  Same for others below.

Reply:
Thanks, I didn't know this! I'll remove them.